### DISCUSSION TITLE
Gizmo status lights set to "connected missing data"

### DISCUSSION THREAD

**Discussion by TheNoodle0** [gizmo-platform/gizmo#38] [Q&A] [answered]
There's no explanation on how to fix this in the guide. Is it a problem with the fms, ds, firmware, broken part, etc.?

**Comment by the-maldridge** [comment]
Given the similarity of timing and question, you don't happen to be with Music City BEST do you?  If you haven't already engaged your hub leadership, you should do so now so they are aware of your status.

Assuming you're referencing [this diagram](https://gizmoplatform.dev/trouble/gizmo.html) the first thing to check would be that the gamepad is connected to the driver's station and the little switch on the bottom is in the 'D' position.

If the switch is in the 'D' position, the next thing to check would be to re-bind the driver's station to the Gizmo by connecting the USB A to B-micro cable between the system processor (the one on the left) and any USB port on the driver's station.  Power both on and then once they have booted (20-30s) press and hold the bootsel button on the Gizmo system processor (left one).  The Gizmo will reboot and reconnect.  This process is described in the manual [here](https://gizmoplatform.dev/startup/firmware.html#bind-the-gizmo-to-a-drivers-station).

If neither of those steps work, we're at the point where its time to consider reinstalling the firmware on both sides to rule that out, but I'd want you to be in contact with your hub before you do that.

**Reply by TheNoodle0** [reply]
I did post this on the forum earlier and I'll reach out to them if this doesn't work.

We did reinstall the firmware (v01.00) thinking that was the issue so right now the field and network lights are flashing.

**Reply by the-maldridge** [reply]
The firmware has 2 version numbers.  You have a v1.0 board which means you need the  gss-1_0_R00-v0.1.3.uf2 firmware file, but you need the most recent release of that, which is v0.1.3.  You can find that file [here](https://github.com/gizmo-platform/firmware/releases/tag/v0.1.3).

**Reply by TheNoodle0** [reply]
I did all that, now it keeps flipping between all four light indicators when connected to the driver station.

**Reply by the-maldridge** [reply, marked as answer]
We have identified a stability issue with v0.1.3 and this release has been retracted.  The most recent qualified release is now v0.1.1.

**Reply by TheNoodle0** [reply]
One of our Gizmos is faulty, we tried it on the other one and it worked. Now we're back to the original issue where we're putting all the code in correctly in both Arduino and Python, but the motors aren't running

**Reply by TheNoodle0** [reply]
The motor runs when I take off the motor controller and plug it in alone, so there's no hardware issue. Arduino has all the correct libraries and doesn't have any issues when compiling. Circuit Python has the Adafruit firmware and Circup.
I've been struggling with this for weeks and I'm about to give up.

**Reply by the-maldridge** [reply]
If you believe your Gizmo is faulty, then you need to reach out to your hub and request the hardware be swapped.

**Reply by TheNoodle0** [reply]
we have 2, one of them doesn't pair to the driver station and the other seems to be working, but hasn't moved anything.

**Reply by gdyoung93** [reply]
Which hub are you with?

**Reply by TheNoodle0** [reply]
I'm not asking how to replace the gizmo; we're already doing that. I want to understand what could be wrong with this so we can actually compete.

**Reply by TheNoodle0** [reply]
If I'm missing something and we replace the gizmo, it's just going to be the same problem all over again

**Reply by gdyoung93** [reply]
Can you provide the hub you are competing with?

**Reply by the-maldridge** [reply]
At this point any further troubleshooting is going to involve pulling the case and debugging at a lower level, something teams are expressly not permitted to do without approval from the hub that issued their hardware.

**Reply by TheNoodle0** [reply]
I'm with team 1043 at Music City

The first one is magically working now and is having the same issue, so either both of them have the same issue, or more probably, I'm missing a step. I'd like to go back through the process we took before jumping to taking everything apart.

**Comment by gdyoung93** [comment]
Can you please contact your hub leadership (juan.rojas@bestinc.org) at the top of your Team Workflow.  They can walk you through steps. You should NOT be taking the Gizmo apart without their blessing and knowledge.
